src/barnetilsyn/BarnetilsynApp.tsx

```diff
@@ -0,0 +1,123 @@
+import {
+  autentiseringsInterceptor,
+  verifiserAtBrukerErAutentisert,
+} from '../utils/autentisering';
+import mockPersonMedBarn from '../mock/mockPerson.json';
+import mockPersonUtenBarn from '../mock/mockPersonUtenBarn.json';
+import mockToggles from '../mock/mockToggles.json';
+import { settLabelOgVerdi } from '../utils/søknad';
+import { standardLabelsBarn } from '../helpers/labels';
+import { useBarnetilsynSøknad } from './BarnetilsynContext';
+import { useToggles } from '../context/TogglesContext';
+import { IPerson } from '../models/person';
+import { Helmet } from 'react-helmet';
+import { erLokaltMedMock } from '../utils/miljø';
+import SøknadsdialogBarnetilsyn from './Søknadsdialog';
+
+const BarnetilsynApp = () => {
+  const [autentisert, settAutentisering] = useState<boolean>(false);
+  const [fetching, settFetching] = useState<boolean>(true);
+  const [error, settError] = useState<boolean>(false);
+  const { settPerson } = usePersonContext();
+  const {
+    søknad,
+    settSøknad,
+    hentMellomlagretBarnetilsyn,
+  } = useBarnetilsynSøknad();
+  const { settToggles, toggles } = useToggles();
+
+  autentiseringsInterceptor();
+
+  useEffect(() => {
+    verifiserAtBrukerErAutentisert(settAutentisering);
+  }, [autentisert]);
+
+  const fetchPersonData = () => {
+    return hentPersonData()
+      .then((response) => {
+        settPerson({
+          type: PersonActionTypes.HENT_PERSON,
+          payload: response,
+        });
+        oppdaterSøknadMedBarn(response, response.barn);
+      })
+      .catch(() => settError(true));
+  };
+
+  const oppdaterSøknadMedBarn = (person: IPerson, barneliste: any[]) => {
+    const barnMedLabels = barneliste.map((barn: any) => {
+      const barnMedLabel = settLabelOgVerdi(barn, standardLabelsBarn);
+      barnMedLabel['ident'] = barnMedLabel['fnr'];
+      delete barnMedLabel.fnr;
+      return barnMedLabel;
+    });
+
+    settSøknad &&
+      settSøknad({ ...søknad, person: { ...person, barn: barnMedLabels } });
+  };
+
+  const fetchToggles = () => {
+    return hentToggles(settToggles).catch((err: Error) => {
+      settError(true);
+    });
+  };
+
+  useEffect(() => {
+    if (erLokaltMedMock()) {
+      settPerson({
+        type: PersonActionTypes.HENT_PERSON,
+        payload: mockPersonUtenBarn,
+      });
+      oppdaterSøknadMedBarn(mockPersonUtenBarn, mockPersonMedBarn.barn);
+      settToggles(mockToggles);
+      settFetching(false);
+      return;
+    }
+    Promise.all([
+      fetchToggles(),
+      fetchPersonData(),
+      hentMellomlagretBarnetilsyn(),
+    ])
+      .then(() => settFetching(false))
+      .catch(() => settFetching(false));
+    // eslint-disable-next-line
+  }, []);
+
+  if (!fetching && autentisert) {
+    if (!error) {
+      return (
+        <>
+          <Helmet>
+            <title>Søknad om barnetilsyn</title>
+          </Helmet>
+
+          {!toggles[ToggleName.send_søknad] && <TestsideInformasjon />}
```

Comment:
Burde vi introdusere andre toggles for barnetilsyn?

Reply:
Det synes jeg høres ut som en bra idé! 💡